packages/component/src/SendBox/ConnectivityStatus.js

```diff
@@ -1,110 +1,102 @@
+    const { children, debounce } = nextProps;
+
+    if (
+      children !== this.props.children
+      || debounce !== this.props.debounce
+    ) {
+      clearTimeout(this.timeout);
 
-    if ( status === 'connectingslow' && connectingSlow === false ) {
-      this.setState( () => ( { connectingSlow: true } ) );
+      this.timeout = setTimeout(() => {
+        this.setState(() => ({
+          renderConnectivity: children,
+          since: Date.now()
+        }));
+      }, Math.max(0, debounce - Date.now() + this.state.since));
     }
   }
 
-  static getDerivedStateFromProps( props, state ) {
-    if ( props !== state ) {
-      state = {
-        connectivityStatus: props.connectivityStatus,
-        connectingSlow: false,
-        language: props.language,
-        styleSet: props.styleSet,
-      }
-    }
-    return state;
+  componentWillUnmount() {
+    clearTimeout(this.timeout);
   }
 
   render() {
-    const { connectivityStatus, styleSet } = this.state;
-    const { status } = connectivityStatus;
-    return (
-      <div
-        aria-live="polite"
-        className={classnames( {
-          [ styleSet.errorNotification ]: ( status === 'error' ),
-          [ styleSet.warningNotification ]: status === 'connectingslow' || status === 'reconnecting',
-          [ styleSet.connectivityNotification ]: status === 'connected' || status === 'uninitialized' || status === 'reconnecting'
-        } )}
-        role="status"
-      >
-        {this.connectivityStatusText()}
-      </div>
-    );
+    const { renderConnectivity } = this.state;
+
+    return typeof renderConnectivity === 'function' ? renderConnectivity() : false;
   }
 }
 
-export default connectToWebChat(
-  ( { connectivityStatus, language, styleSet } ) => ( { connectivityStatus, language, styleSet } )
-)( connectConnectivityAlert )
+const connectConnectivityStatus = (...selectors) => connectToWebChat(
+  ({ connectivityStatus, language }) => ({ connectivityStatus, language }),
+  ...selectors
+)
+
+export default connectConnectivityStatus(
+  ({ styleSet }) => ({ styleSet })
+)(
+  ({ connectivityStatus, language, styleSet }) =>
+    <div
+      aria-live="polite"
+      className={
+        classNames({
+          [styleSet.errorNotification]:
+            connectivityStatus === 'error'
+            || connectivityStatus === 'notconnected',
+          [styleSet.warningNotification]:
+            connectivityStatus === 'connectingslow',
+          [styleSet.connectivityNotification]:
+            connectivityStatus === 'uninitialized'
+            || connectivityStatus === 'connected'
+            || connectivityStatus === 'reconnected'
+            || connectivityStatus === 'reconnecting'
+        })
+      }
+      role="status"
+    >
+      <ConnectivityDebounce
+        debounce={ (connectivityStatus === 'uninitialized' || connectivityStatus === 'error') ? 0 : 400 }>
+        { () =>
+          connectivityStatus === 'connectingslow' ?
+            <React.Fragment>
```

Comment:
Are you able to render a single React.Fragment outside the chained ternary like this?
```javascript
() => {
  <React.Fragment>
    {
     connectivityStatus === 'connectingslow' ?
       <WarningNotificationIcon />
     : (connectivityStatus === 'error' || connectivityStatus === 'notconnected') ?
       <ErrorNotificationIcon />
     : connectivityStatus === 'uninitialized' ?
       <SpinnerAnimation />
       { localize('INITIAL_CONNECTION_NOTIFICATION', language) }
    }
 </React.Fragment>
}
```

Reply:
I don't think this will work because each status also has different text to be localized, and you can't return multiple children, only one. 

this gives errors:

```() => {
  <React.Fragment>
    {
     connectivityStatus === 'connectingslow' ?
       <WarningNotificationIcon />
       { localize('SLOW_CONNECTION_NOTIFICATION', language) }
     : (connectivityStatus === 'error' || connectivityStatus === 'notconnected') ?
       <ErrorNotificationIcon />
       { localize('FAILED_CONNECTION_NOTIFICATION', language) }
     : connectivityStatus === 'uninitialized' ?
       <SpinnerAnimation />
       { localize('INITIAL_CONNECTION_NOTIFICATION', language) }
    }
 </React.Fragment>
}```

Playing around with parenthesis didn't get rid of the errors. Let me know if I should try something else. :)